src/ConstrainedAstrometryModel.cc

```diff
@@ -46,60 +48,62 @@ mappings.*/
-            }
+            _visitMap[visit] = std::make_shared<SimplePolyMapping>(GtransfoLin(), GtransfoPoly(visitDegree));
         }
         auto chipp = _chipMap.find(chip);
         if (chipp == _chipMap.end()) {
-            const Frame &frame = im.getImageFrame();
-
-            GtransfoPoly pol(chipDegree);
-
-            if (initFromWCS) {
-                pol = GtransfoPoly(im.getPix2TangentPlane(), frame, chipDegree);
+            auto center = ccdImage->getDetector()->getCenter(afw::cameraGeom::FOCAL_PLANE);
+            double radius2 = std::pow(center.getX(), 2) + std::pow(center.getY(), 2);
+            if (radius2 < minRadius2) {
+                minRadius2 = radius2;
+                constrainedChip = chip;
             }
-            GtransfoLin shiftAndNormalize = normalizeCoordinatesTransfo(frame);
 
-            _chipMap[chip] = std::unique_ptr<SimplePolyMapping>(
-                    new SimplePolyMapping(shiftAndNormalize, pol * shiftAndNormalize.invert()));
+            auto pixelsToFocal =
+                    im.getDetector()->getTransform(afw::cameraGeom::PIXELS, afw::cameraGeom::FOCAL_PLANE);
+            const Frame &frame = im.getImageFrame();
+            // construct the chip gtransfo by approximating the pixel->Focal afw::geom::Transform.
+            GtransfoPoly pol = GtransfoPoly(pixelsToFocal, frame, chipDegree);
+            GtransfoLin shiftAndNormalize = normalizeCoordinatesTransfo(frame);
+            _chipMap[chip] =
+                    std::make_shared<SimplePolyMapping>(shiftAndNormalize, pol * shiftAndNormalize.invert());
         }
     }
+
+    // Hold the "central" chip map fixed and don't fit it, to remove a degeneracy.
+    _chipMap.at(constrainedChip)->setToFit(false);
+
     // now, second loop to set the mappings of the CCdImages
     for (auto &ccdImage : ccdImageList) {
         const CcdImage &im = *ccdImage;
         auto visit = im.getVisit();
         auto chip = im.getCcdId();
+
         // check that the chip_indexed part was indeed assigned
         // (i.e. the reference visit was complete)
         if (_chipMap.find(chip) == _chipMap.end()) {
             LOGLS_WARN(_log, "Chip " << chip << " is missing in the reference exposure, expect troubles.");
             GtransfoLin norm = normalizeCoordinatesTransfo(im.getImageFrame());
-            _chipMap[chip] =
-                    std::unique_ptr<SimplePolyMapping>(new SimplePolyMapping(norm, GtransfoPoly(chipDegree)));
+            _chipMap[chip] = std::make_shared<SimplePolyMapping>(norm, GtransfoPoly(chipDegree));
         }
-        _mappings[ccdImage->getHashKey()] = std::unique_ptr<TwoTransfoMapping>(
-                new TwoTransfoMapping(_chipMap[chip].get(), _visitMap[visit].get()));
+        _mappings[ccdImage->getHashKey()] =
+                std::make_unique<TwoTransfoMapping>(_chipMap[chip], _visitMap[visit]);
     }
     LOGLS_INFO(_log, "Constructor got " << _chipMap.size() << " chip mappings and " << _visitMap.size()
```

Comment:
Hmmm... To me it feels like an info log, because it's the information you need to know to understand the relative sizes of the model components. I've been erring on the side of being slightly more verbose with jointcal logs (particularly since debug results in *very* verbose logs). Unfortunately, the dev docs don't provide much guidance on how to choose a log level...

Reply:
"Constructor got" screams debug level.

Reply:
That's fair. What if I reworded it to read: `Constrained astrometry model has X chip mappings ...`?

Reply:
Yes, that's polished as well as useful, so info level would be fine.